frontend/packages/@depmap/types/src/BreadboxErrorType.ts

```diff
@@ -0,0 +1,56 @@
+// should match enums defined in custom_http_exception.py in breadbox
+type ErrorType =
+  | "DATASET_NOT_FOUND"
+  | "FEATURE_NOT_FOUND"
+  | "SAMPLE_NOT_FOUND"
+  | "DIMENSION_TYPE_NOT_FOUND"
+  | "LARGE_DATASET_READ";
+
+export interface ErrorDetail {
+  error_type: ErrorType;
+  message: string;
+}
+
+/* Typing for custom exceptions thrown by Breadbox */
+interface BreadboxCustomException {
+  detail: string | ErrorDetail; // also string type for backwards compatibility.
+}
+
+export function instanceOfBreadboxCustomException(
+  object: any
+): object is BreadboxCustomException {
+  return typeof object === "object" && object !== null && "detail" in object;
+}
+
+export function instanceOfErrorDetail(object: any): object is ErrorDetail {
+  return (
+    typeof object === "object" &&
+    object !== null &&
+    "error_type" in object &&
+    "message" in object
+  );
+}
+
+/* Custom Error class with error type */
+export class ErrorTypeError extends Error {
+  name: ErrorType;
+
+  message: string;
+
+  cause?: any;
+
+  constructor({
+    name,
+    message,
+    cause,
+  }: {
+    name: ErrorType;
+    message: string;
+    cause?: any;
+  }) {
+    super();
```

Comment:
`super()` should be called with `message` so it shows up in stack traces.

<img width="462" height="416" alt="super" src="https://github.com/user-attachments/assets/47690c42-e2ab-4bad-bedc-28ea4ee1d2d2" />